polygeist/tools/cgeist/Test/Verification/shr.cl

```diff
@@ -0,0 +1,234 @@
+// CHECK-NEXT:      %[[VAL_3:.*]] = arith.extsi %[[VAL_0]] : i8 to i32
+// CHECK-NEXT:      %[[VAL_4:.*]] = arith.extsi %[[VAL_1]] : i8 to i32
+// CHECK-NEXT:      %[[VAL_5:.*]] = arith.andi %[[VAL_4]], %[[VAL_2]] : i32
+// CHECK-NEXT:      %[[VAL_6:.*]] = arith.shrsi %[[VAL_3]], %[[VAL_5]] : i32
+// CHECK-NEXT:      %[[VAL_7:.*]] = arith.trunci %[[VAL_6]] : i32 to i8
+// CHECK-NEXT:      return %[[VAL_7]] : i8
+// CHECK-NEXT:    }
+char shr_i8(char a, char b) {
+  return a >> b;
+}
+
+
+// CHECK-LABEL:   func.func @shr_i16(
+// CHECK-SAME:                       %[[VAL_0:.*]]: i16,
+// CHECK-SAME:                       %[[VAL_1:.*]]: i16) -> i16
+// CHECK-NEXT:      %[[VAL_2:.*]] = arith.constant 31 : i32
+// CHECK-NEXT:      %[[VAL_3:.*]] = arith.extsi %[[VAL_0]] : i16 to i32
+// CHECK-NEXT:      %[[VAL_4:.*]] = arith.extsi %[[VAL_1]] : i16 to i32
+// CHECK-NEXT:      %[[VAL_5:.*]] = arith.andi %[[VAL_4]], %[[VAL_2]] : i32
+// CHECK-NEXT:      %[[VAL_6:.*]] = arith.shrsi %[[VAL_3]], %[[VAL_5]] : i32
+// CHECK-NEXT:      %[[VAL_7:.*]] = arith.trunci %[[VAL_6]] : i32 to i16
+// CHECK-NEXT:      return %[[VAL_7]] : i16
+// CHECK-NEXT:    }
+short shr_i16(short a, short b) {
+  return a >> b;
+}
+
+
+// CHECK-LABEL:   func.func @shr_i32(
+// CHECK-SAME:                       %[[VAL_0:.*]]: i32,
+// CHECK-SAME:                       %[[VAL_1:.*]]: i32) -> i32
+// CHECK-NEXT:      %[[VAL_2:.*]] = arith.constant 31 : i32
+// CHECK-NEXT:      %[[VAL_3:.*]] = arith.andi %[[VAL_1]], %[[VAL_2]] : i32
+// CHECK-NEXT:      %[[VAL_4:.*]] = arith.shrsi %[[VAL_0]], %[[VAL_3]] : i32
+// CHECK-NEXT:      return %[[VAL_4]] : i32
+// CHECK-NEXT:    }
+int shr_i32(int a, int b) {
+  return a >> b;
+}
+
+
+// CHECK-LABEL:   func.func @shr_i64(
+// CHECK-SAME:                       %[[VAL_0:.*]]: i64,
+// CHECK-SAME:                       %[[VAL_1:.*]]: i64) -> i64
+// CHECK-NEXT:      %[[VAL_2:.*]] = arith.constant 63 : i64
+// CHECK-NEXT:      %[[VAL_3:.*]] = arith.andi %[[VAL_1]], %[[VAL_2]] : i64
+// CHECK-NEXT:      %[[VAL_4:.*]] = arith.shrsi %[[VAL_0]], %[[VAL_3]] : i64
+// CHECK-NEXT:      return %[[VAL_4]] : i64
+// CHECK-NEXT:    }
+long shr_i64(long a, long b) {
+  return a >> b;
+}
+
+
+// CHECK-LABEL:   func.func @shr_ui8(
+// CHECK-SAME:                       %[[VAL_0:.*]]: i8,
+// CHECK-SAME:                       %[[VAL_1:.*]]: i8) -> i8
+// CHECK-NEXT:      %[[VAL_2:.*]] = arith.constant 31 : i32
+// CHECK-NEXT:      %[[VAL_3:.*]] = arith.extui %[[VAL_0]] : i8 to i32
+// CHECK-NEXT:      %[[VAL_4:.*]] = arith.extui %[[VAL_1]] : i8 to i32
+// CHECK-NEXT:      %[[VAL_5:.*]] = arith.andi %[[VAL_4]], %[[VAL_2]] : i32
+// CHECK-NEXT:      %[[VAL_6:.*]] = arith.shrsi %[[VAL_3]], %[[VAL_5]] : i32
+// CHECK-NEXT:      %[[VAL_7:.*]] = arith.trunci %[[VAL_6]] : i32 to i8
+// CHECK-NEXT:      return %[[VAL_7]] : i8
+// CHECK-NEXT:    }
+unsigned char shr_ui8(unsigned char a, unsigned char b) {
+  return a >> b;
+}
```

Comment:
Operations with unsigned operands needing an extension are signed, as the extension is to signed `int`.